### PR TITLE
Bump dependencies.

### PR DESCRIPTION
**Pull request by adlerjohn**
Dependencies
- `fuel-asm` to `0.2`
- `fuel-crypto` to `0.3`
- `fuel-gql-client` to `0.4`
- `fuel-tx` to `0.6`
- `fuel-vm` to `0.5`

CI
- `fuel-core` to `0.4.1`

### CODE REVIEW

**Comment by adlerjohn** [comment]
Actually renaming `Color` to `AssetId` will be left to a future PR.

File: Cargo.lock [hunk 1]
```diff
@@ -995,10 +1001,10 @@ dependencies = [
  "clap 3.1.2",
  "clap_complete",
  "dirs 3.0.2",
- "fuel-asm",
+ "fuel-asm 0.2.0",
  "fuel-gql-client",
- "fuel-tx",
- "fuel-vm",
+ "fuel-tx 0.6.0",
+ "fuel-vm 0.5.0",
```
Comment (mitchmindtree) [inline Cargo.lock:1007]: When you see the `Cargo.lock` start to specify the version for dependencies like this, it's a hint that there might be two crates in the workspace that have different versions of the same dependency, or that we may have forgotten to update one or more crates' dependencies.

In this case, it looks like `sway-core`'s `fuel-asm` and `fuel-vm` deps might have been missed?

**Review by mitchmindtree — CHANGES_REQUESTED** [review]

File: Cargo.lock (hunk 1 above)
Comment (adlerjohn) [reply Cargo.lock:1007]: I was wondering why sometimes there were numbers and sometimes not!

Comment (adlerjohn) [reply Cargo.lock]: Fixed in [01cafb3](https://github.com/FuelLabs/sway/pull/886/commits/01cafb3f7e5d6968e473dea025bcb909d23072c2)

**Review by mitchmindtree — APPROVED** [review]

**Pull request merged**